actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -266,13 +273,17 @@ pub fn call_generic<RT: Runtime>(
 
                         // and then invoke self with delegate; readonly context is sticky
                         let params = DelegateCallParams { code, input: input_data.into() };
-                        system.send_with_gas(
+                        system.send_generalized(
                             &system.rt.message().receiver(),
                             Method::InvokeContractDelegate as u64,
                             RawBytes::serialize(&params)?,
                             TokenAmount::from(&value),
                             effective_gas_limit(system, gas),
-                            system.readonly,
+                            if system.readonly {
+                                SendFlags::READ_ONLY
+                            } else {
+                                SendFlags::default()
+                            },
```

Comment:
This isn't necessary, the FVM itself will propagate the read-only status.

Reply:
yep, my issue earlier was that I had modified the test vm to _not_ propagate, now fixed though.